crates/red_knot_python_semantic/src/types.rs

```diff
@@ -605,8 +609,6 @@ impl<'db> Type<'db> {
             return false;
         }
         match (self, target) {
-            (Type::Unknown | Type::Any | Type::Todo(_), _) => false,
-            (_, Type::Unknown | Type::Any | Type::Todo(_)) => false,
```

Comment:
I removed these since they (and the new `Subclass(Any)` form) are handled by the `is_fully_static` checks above.